### PR TITLE
Query by hash prefix

### PR DESCRIPTION
**Pull request by Joe-Dowd**
Continued from the work by https://github.com/Zegnat in https://github.com/ajayyy/SponsorBlockServer/pull/86

### CODE REVIEW

**Comment by Joe-Dowd** [comment]
branched off https://github.com/ajayyy/SponsorBlockServer/pull/114 so merge that first,

**Comment by Joe-Dowd** [comment]
Video id is hashed using 1 pass sha256

Interface:
`GET /api/skipSegments/:hashPrefix?categories=["sponsor," "intro"]`
response:
``` 
[
  {
    "videoID": "blah",
    "hash": "jhdshdjsd",
    segments: ... /* Regular GET skipSegments response */
  }
]
```

File: src/utils/hashPrefixTester.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const config = require('../config.js');
+
+const minimumPrefix = config.minimumPrefix || '3';
+const maximumPrefix = config.maximumPrefix || '32'; // Half the hash.
```
Comment (ajayyy) [inline src/utils/hashPrefixTester.js:4]: Is there a reason to have a maximum?

Comment (Joe-Dowd) [reply src/utils/hashPrefixTester.js:4]: probably not but it doesn't hurt

Comment (Zegnat) [reply src/utils/hashPrefixTester.js:4]: Having a maximum is more to limit clients from doing the wrong thing. Discourage sending too much of the hash, as that would just be tossing the anonimity layer out again.

**Pull request merged**